### PR TITLE
Add types for properties

### PR DESCRIPTION
**Pull request by yuvalabou**
Add typing for properties - not all as we do not check full annotations on this package, and its just part of PR https://github.com/home-assistant/core/pull/51681

### CODE REVIEW

**Comment by KapJI** [comment]
Please note that in order to use these type hints in HA, typing information needs to be exported by adding `py.typed` marker. This requires having all functions type annotated, otherwise they won't be found by mypy.

File: hole/__init__.py [hunk 1]
```diff
@@ -34,7 +35,7 @@ def __init__(
         self.host = host
         self.location = location
         self.api_token = api_token
-        self.data = {}
+        self.data: dict[str, Any] = {}
```
Comment (KapJI) [inline hole/__init__.py:38]: This requires `from __future__ import annotations`.
```suggestion
        self.data: dict[str, Any] = {}
```

Comment (yuvalabou) [reply hole/__init__.py:39]: correct - however HA uses 3.7 at its minimum so I don't think thats a problem

File: hole/__init__.py [hunk 2]
```diff
@@ -17,14 +19,14 @@ class Hole(object):
...
+        host: str,
         loop,
         session,
-        location="admin",
-        tls=False,
-        verify_tls=True,
+        location: str="admin",
+        tls: bool=False,
+        verify_tls: bool=True,
         api_token=None,
```
Comment (KapJI) [inline hole/__init__.py:28]:
```suggestion
        api_token: str | None = None,
```

File: hole/__init__.py [hunk 3]
```diff
@@ -17,14 +19,14 @@ class Hole(object):
 
     def __init__(
         self,
-        host,
+        host: str,
         loop,
         session,
```
Comment (KapJI) [inline hole/__init__.py:24]:
```suggestion
        loop: asyncio.events.AbstractEventLoop,
        session: aiohttp.ClientSession,
```

**Review by KapJI — COMMENTED** [review]
Let's add missing types.

File: hole/__init__.py (hunk 2 above)
Comment (yuvalabou) [reply hole/__init__.py]: thanks

**Comment by KapJI** [comment]
It'll fail on Python 3.6: https://github.com/home-assistant-ecosystem/python-hole/blob/master/.github/workflows/python.yml
Also mypy is not being run as part of CI. And I think your code is not formatted with black.

**Comment by yuvalabou** [comment]
> It'll fail on Python 3.6: https://github.com/home-assistant-ecosystem/python-hole/blob/master/.github/workflows/python.yml
> Also mypy is not being run as part of CI. And I think your code is not formatted with black.

Do we really need support for python 3.6? Most common I have seen support for the latest 3 version which are 3.7,3.8, and 3.9

**Comment by fabaff** [comment]
Python 3.7 and even Python 3.8 can be skipped, from my point of view, as distributions are starting to move to 3.10.

**Pull request closed without merging**